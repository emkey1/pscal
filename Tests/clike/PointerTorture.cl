int main() {
    int values[4];
    int i;

    for (i = 0; i < 4; i = i + 1) {
        values[i] = i * 10;
    }

    int *p0 = &values[3];
    int *p1 = &values[2];
    int *p2 = &values[1];
    int *p3 = &values[0];

    printf("%d %d %d %d\n", *p0, *p1, *p2, *p3);

    *p1 = *p1 + 5;
    *p3 = 99;

    int *tmp = p0;
    p0 = p3;
    p3 = tmp;

    printf("%d %d %d %d\n", *p0, *p1, *p2, *p3);

    *p2 = *p0;

    for (i = 0; i < 4; i = i + 1) {
        printf("%d\n", values[i]);
    }

    return 0;
}
