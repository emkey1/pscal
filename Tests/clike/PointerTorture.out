30 20 10 0
99 25 10 30
99
99
25
30
